integration_tests/dataplane/mymac/mymac_test.go

```diff
@@ -189,65 +186,147 @@ func clearMyMac(t *testing.T, dut *ondatra.DUTDevice) error {
 		return nil
 	}
 	for _, oid := range oids {
-		mmc := saipb.NewMyMacClient(conn)
 		if _, err := mmc.RemoveMyMac(context.Background(), &saipb.RemoveMyMacRequest{
 			Oid: oid,
 		}); err != nil {
 			return fmt.Errorf("Failed to remove MyMac: %+v", err)
 		}
+		t.Logf("Removed OID: %d", oid)
 	}
 	t.Logf("MyMac table cleared.")
 	return nil
 }
 
 // restoreMyMac restores the rule to allow all traffic to send to L3.
-func restoreMyMac(t *testing.T, dut *ondatra.DUTDevice) error {
+func restoreMyMac(t *testing.T, mmc saipb.MyMacClient) (uint64, error) {
 	// Allow all traffic to L3 processing.
-	conn := dataplaneConn(t, dut)
-	mmc := saipb.NewMyMacClient(conn)
-	_, err := mmc.CreateMyMac(context.Background(), &saipb.CreateMyMacRequest{
+	resp, err := mmc.CreateMyMac(context.Background(), &saipb.CreateMyMacRequest{
 		Switch:         1,
 		Priority:       proto.Uint32(1),
 		MacAddress:     []byte{0, 0, 0, 0, 0, 0},
 		MacAddressMask: []byte{0, 0, 0, 0, 0, 0},
 	})
 	if err != nil {
-		return err
+		return 0, err
 	}
 	t.Logf("MyMac table restored.")
+	return resp.GetOid(), nil
+}
+
+// printMyMacEntries prints the entries in the MyMac table.
+func printMyMacEntries(t *testing.T, sc saipb.SwitchClient, mmc saipb.MyMacClient) error {
+	var err error
+	req := &saipb.GetSwitchAttributeRequest{
+		Oid:      1,
+		AttrType: []saipb.SwitchAttr{saipb.SwitchAttr_SWITCH_ATTR_MY_MAC_LIST},
+	}
+	resp, err := sc.GetSwitchAttribute(context.Background(), req)
+	if err != nil {
+		return err
+	}
+	oids := resp.GetAttr().MyMacList
+	if oids == nil {
+		return nil
+	}
+	for _, oid := range oids {
+		resp2, err := mmc.GetMyMacAttribute(context.Background(), &saipb.GetMyMacAttributeRequest{
+			Oid: oid,
+		})
+		if err != nil {
+			return err
+		}
+		t.Logf(">>>>>> MyMac[%d]: %+v", oid, resp2.GetAttr())
+	}
 	return nil
 }
 
+// entry contains the MyMac request information.
+type entry struct {
+	priority    uint32
+	macAddr     string
+	macAddrMask []byte
+}
+
+type myMacTest struct {
+	desc       string
+	clearMyMac bool
+	dstMAC     string
+	entries    []entry
+	passed     bool
```

Comment:
nit: rename passed to wantTraffic (or something like that wantNoDrop etc) since all tests should pass

Reply:
done